### ISSUE TITLE
Playbook for running cassandra cluster roles

### ISSUE THREAD

**Issue by praveen049** [andrewrothstein/ansible-cassandra-cluster#2]
Hi
  I am using the below playbook to run the cassandra cluster role:
```
- hosts: cassandra
  roles:
  - role: andrewrothstein.cassandra-cluster
    cassandra_network_iface: eth0
    cassandra_seeds: 192.168.0.15
    ty: group
      name: cassandra-cluster
```

but getting this error.
```
fatal: [192.168.0.24]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: {% if cassandra_seeds.ty == \"static\" %}{{cassandra_seeds | join(\",\")}}{% else %}{% for host in groups[cassandra_groupname] %}{{hostvars[host][\"ansible_\" + hostvars[host][\"cassandra_network_iface\"]][\"ipv4\"][\"address\"]}}{% if not loop.last %},{% endif %}{% endfor %}{% endif %}: 'ansible.parsing.yaml.objects.AnsibleUnicode object' has no attribute 'ty'"}
fatal: [192.168.0.25]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: {% if cassandra_seeds.ty == \"static\" %}{{cassandra_seeds | join(\",\")}}{% else %}{% for host in groups[cassandra_groupname] %}{{hostvars[host][\"ansible_\" + hostvars[host][\"cassandra_network_iface\"]][\"ipv4\"][\"address\"]}}{% if not loop.last %},{% endif %}{% endfor %}{% endif %}: 'ansible.parsing.yaml.objects.AnsibleUnicode object' has no attribute 'ty'"}
fatal: [192.168.0.15]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: {% if cassandra_seeds.ty == \"static\" %}{{cassandra_seeds | join(\",\")}}{% else %}{% for host in groups[cassandra_groupname] %}{{hostvars[host][\"ansible_\" + hostvars[host][\"cassandra_network_iface\"]][\"ipv4\"][\"address\"]}}{% if not loop.last %},{% endif %}{% endfor %}{% endif %}: 'ansible.parsing.yaml.objects.AnsibleUnicode object' has no attribute 'ty'"}
```

The hosts file is
```
[cassandra]
192.168.0.24
192.168.0.25
192.168.0.15
```

The command to run
`ansible-playbook -i hosts -u cassandra cluster_cassandra.yml  --extra-vars "ansible_sudo_pass=pass"
`
I suspect that the syntax in the playbook is wrong. Can you please help clarify what is the error ?

thanks

**Comment by andrewrothstein** [comment]
take a look at [defaults/main.yml](https://github.com/andrewrothstein/ansible-cassandra-cluster/blob/master/defaults/main.yml). Do you want to get the seed IPs from Ansible facts or to specify them as role vars?

```yml
cassandra_seeds:
  ty: static
  seeds:
    - 192.168.0.15
```

should work to specify the seed IPs statically as a role var. Please upgrade to the latest version of this role as I just made a couple of bug fixes (v2.0.1).

**Comment by praveen049** [comment]
Thanks for the inputs. I have now changed the playbook as i would like to include them in the playbook vars.
This is the current configuration
```
- hosts: cassandra
  roles:
  - role: andrewrothstein.cassandra-cluster
    cassandra_network_iface: eth0
    cassandra_seeds:
     ty: static
     seeds:
       - 192.168.0.20
       - 192.168.0.26
       - 192.168.0.27
    name: cassandra-cluster
  environment:
    http_proxy: http://10.158.100.3:8080/
    https_proxy: http://10.158.100.3:8080/
```

With this the playbook execution is going through correctly.

On the target nodes, i login to the cassandra and check the status but the status is down.
```
[cassandra@cassandra-3:~$ sudo service cassandra status
[sudo] password for cassandra: 
● cassandra.service - Cassandra
   Loaded: loaded (/etc/systemd/system/cassandra.service; enabled; vendor preset: enabled)
   Active: inactive (dead) (Result: exit-code) since Thu 2018-03-22 01:47:47 UTC; 1min 26s ago
  Process: 3756 ExecStart=/usr/local/cassandra/bin/cassandra -f -R -p /usr/local/cassandra/run/cassandra.pid (code=exited, status=1/FAILURE)
  Process: 3753 ExecStartPre=/usr/bin/mkdir -p /usr/local/cassandra/data/saved_caches (code=exited, status=203/EXEC)
  Process: 3750 ExecStartPre=/usr/bin/mkdir -p /usr/local/cassandra/data/commitlog (code=exited, status=203/EXEC)
  Process: 3747 ExecStartPre=/usr/bin/mkdir -p /usr/local/cassandra/data/data (code=exited, status=203/EXEC)
  Process: 3744 ExecStartPre=/usr/bin/mkdir -p /usr/local/cassandra/data/hints (code=exited, status=203/EXEC)
 Main PID: 3756 (code=exited, status=1/FAILURE)

Mar 22 01:47:46 cassandra-3 systemd[1]: cassandra.service: Unit entered failed state.
Mar 22 01:47:46 cassandra-3 systemd[1]: cassandra.service: Failed with result 'exit-code'.
Mar 22 01:47:47 cassandra-3 systemd[1]: cassandra.service: Service hold-off time over, scheduling restart.
Mar 22 01:47:47 cassandra-3 systemd[1]: Stopped Cassandra.
Mar 22 01:47:47 cassandra-3 systemd[1]: cassandra.service: Start request repeated too quickly.
Mar 22 01:47:47 cassandra-3 systemd[1]: Failed to start Cassandra.](url)

```
sudo service cassandra start does not help.

Also start as 
`cassandra `
gives the error
```
WARN  [main] 2018-03-22 01:50:41,869 StartupChecks.java:332 - Directory /usr/local/cassandra/data/data doesn't exist
ERROR [main] 2018-03-22 01:50:41,869 CassandraDaemon.java:708 - Has no permission to create directory /usr/local/cassandra/data/data
```

So, i need to manually set the permission of the directory in usr local
`cassandra@cassandra-3:/usr/local$ sudo chown -R cassandra:cassandra apache-cassandra-3.11.2/
`
and after that 
cassandra
`cassandra@cassandra-1:/usr/local$ cassandra`
start well. and node tool shows the status correctly.
cqlsh also work fine.

So to summarize, the roles works almost perfectly. 
If the service can be start automatically and if service status can work it will be perfect.

Thank you very much.

**Comment by andrewrothstein** [comment]
See the defaults.yml file. The creation of the data directory is optional since some folks might want to mount the data directory in from a volume somewhere else. Add the following role var and you'll be GTG.

```yml
cassandra_data_file_directories:
  - /usr/local/cassandra/data/data
```

The `cassandra_activate` role var defaults to True so executing the role should activate the corresponding systemd service.
